src/python/grapl-dgraph-ttl/app.py

```diff
@@ -0,0 +1,91 @@
+import datetime
+import logging
+from typing import Dict, Iterable, Iterator, Optional
+
+from chalice import Chalice
+
+from grapl_analyzerlib.grapl_client import (
+    GraphClient,
+    LocalMasterGraphClient,
+    MasterGraphClient,
+)
+
+IS_LOCAL = bool(os.environ.get("IS_LOCAL", False))
+GRAPL_DGRAPH_TTL_S = int(os.environ.get("GRAPL_DGRAPH_TTL_S", "-1"))
+GRAPL_LOG_LEVEL = os.environ.get("GRAPL_LOG_LEVEL", "ERROR")
+GRAPL_TTL_DELETE_BATCH_SIZE = int(os.environ.get("GRAPL_TTL_DELETE_BATCH_SIZE", "100"))
+
+app = Chalice(app_name="grapl-dgraph-ttl")
+app.log.setLevel(GRAPL_LOG_LEVEL)
+
+
+def query_batch(
+    client: GraphClient,
+    batch_size: int,
+    ttl_cutoff_s: int,
+    last_uid: Optional[str] = None,
+) -> Dict[str, Any]:
+    after = "" if last_uid is None else f" after: {last_uid}"
+    paging = f"first: {batch_size}{after}"
+    query = f"""
+    {{
+        q(func: le(last_index_time, {ttl_cutoff_s}) {paging}) {{
+            uid,
+            expand(_all_) {{ expand(_all_) }}
+        }}
+    }}
+    """
+
+    txn = client.txn()
+    try:
+        return txn.query(query)
+    finally:
+        txn.discard()
+
+
+def calculate_ttl_cutoff_s(now: datetime.datetime, ttl_s: int) -> int:
+    delta = datetime.timedelta(seconds=ttl_s)
+    cutoff = now - delta
+    return int(cutoff.timestamp())
+
+
+def expired_node_uids(
+    client: GraphClient, now: datetime.datetime, ttl_s: int, batch_size: int
+) -> Iterator[Iterable[str]]:
```

Comment:
You're not going to want to return an Iterable[str]. Probably instead something like:

`Iterable[Union[str, Tuple[str, str, str]]]`

Since we're either returning the node uid to delete, or the reverse edges to delete. So far this code looks like it only deletes the uids, but we do need to delete the reverse edges too, or they'll hang.